### DISCUSSION TITLE
Deprecated NULL in 1st arg implode

### DISCUSSION THREAD

**Discussion by T35** [PHPOffice/PhpSpreadsheet#3192] [General]
\PhpOffice\PhpSpreadsheet\Calculation\TextData\Concatenate::TEXTJOIN
...
88 $returnValue = ($returnValue !== '') ? $returnValue : implode($delimiter, $aArgs);
...
(!) $delimiter = NULL if the function in the table looks like this: =TEXTJOIN(;;$B13:$E13)
PHP 8.1.11
LibreOffice

**Comment by MarkBaker** [comment]
I do love the consistency of Microsoft documentation. The [official documented signature](https://support.microsoft.com/en-us/office/textjoin-function-357b449a-ec91-49d0-80c3-0e8fc845691c) for `TEXTJOIN()` is:
```
TEXTJOIN(delimiter, ignore_empty, text1, [text2], …)
```
argument | description |
---|---|
delimiter (required) | A text string, either empty, or one or more characters enclosed by double quotes, or a reference to a valid text string. If a number is supplied, it will be treated as text.
ignore_empty (required) | If TRUE, ignores empty cells.
text1 (required) | Text item to be joined. A text string, or array of strings, such as a range of cells.
[text2, ...] (optional) | Additional text items to be joined. There can be a maximum of 252 text arguments for the text items, including text1. Each can be a text string, or array of strings, such as a range of cells.

So `delimiter` and `ignore_empty` should be mandatory according to that.

But Excel itself does allow
```
=TEXTJOIN(;;$B13:$E13)
```
like LibreOffice

---

> required
> /rɪˈkwʌɪəd/
> adjective
>>officially compulsory, or otherwise considered essential; indispensable.

![image](https://user-images.githubusercontent.com/770298/203027118-c3c0a2cb-b39a-4fef-b54b-e6713cac4bd2.png)

**Comment by MarkBaker** [comment]
[PR #3197](https://github.com/PHPOffice/PhpSpreadsheet/pull/3197)
